v2/protocol/http/protocol.go

```diff
@@ -390,6 +391,11 @@ func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		}
 
 		rw.WriteHeader(status)
+		if errMsg != "" {
```

Comment:
I believe go does not like to write any body after writing the status header. Can you flip these lines?

Reply:
Thanks for your advice @n3wscott . It seems that we have to write body after writing the header. In Go source code https://github.com/golang/go/blob/dev.boringcrypto.go1.18/src/net/http/server.go#L1600, which shows that it will automatically set header when calls `Write`, and in https://github.com/golang/go/blob/dev.boringcrypto.go1.18/src/net/http/server.go#L1146, which shows that after any ` WriteHeader` called, the flag will be set to `true`, and in https://github.com/golang/go/blob/dev.boringcrypto.go1.18/src/net/http/server.go#L1140, which checks the flag and if it was the `true`, the new `StatusCode` will be ignored. I was wondering if my understanding was right, hope your feedback. : )